Source/WebKit/NetworkProcess/curl/WebSocketTaskCurl.cpp

```diff
@@ -345,15 +345,15 @@ std::optional<String> WebSocketTask::receiveFrames(Function<void(WebCore::WebSoc
                 m_continuousFrameOpCode = frame.opCode;
             }
 
-            m_continuousFrameData.append(frame.payload, frame.payloadLength);
+            m_continuousFrameData.append(frame.payload);
 
             if (frame.final) {
                 callback(m_continuousFrameOpCode, m_continuousFrameData.data(), m_continuousFrameData.size());
                 m_hasContinuousFrame = false;
                 m_continuousFrameData.clear();
             }
         } else
-            callback(frame.opCode, frame.payload, frame.payloadLength);
+            callback(frame.opCode, frame.payload.data(), frame.payload.size());
```

Comment:
It would be nice if the `callback` could be modified to expect a `std::span<>`.

Reply:
I agree but since this is Curl code I couldn't build locally, I mostly focused on keeping things building.

Reply:
Anyway, you've convinced me. I'll fix the callback :)